tests/test/projection.rs

```diff
@@ -76,6 +76,7 @@ fn normalize_basic() {
             "Unique"
         }
 
+        /* TODO: this doesn't seem to be correct, actually
```

Comment:
Right, so the problem is that if `U = Vec<T>`, then we have to prove `Normalized(<T as Iterator>::Item -> T)`, which of course we can't do. I wonder why the SLG solver considers this ambiguous, actually? Maybe worth investigating?

I think I'd rather either delete the test or make it `yields[SLGSolver]` and open a FIXME issue to potentially investigate why it behaves this way.

Reply:
Also a `yields[Recursive]` to record the recursive behavior.

Reply:
I think I misunderstood the test -- the recursive solver actually also gives ambiguous. The comment is wrong though, the alternative isn't `Vec<T>`, but `Vec<Iterator::Item<T>>`. I added back the test and fixed the comment